build.rs

```diff
@@ -3,40 +3,79 @@ extern crate cc;
 use std::env;
 
 fn main() {
+    // Check whether we can use 64-bit compilation
+    let use_64bit_compilation = if env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap() == "64" {
+        let check = cc::Build::new()
+            .file("depend/check_uint128_t.c")
+            .cargo_metadata(false)
+            .try_compile("check_uint128_t")
+            .is_ok();
+        if !check {
+            println!("cargo:warning=Compiling in 32-bit mode on a 64-bit architecture due to lack of uint128_t support.");
+        }
+        check
+    } else {
+        false
+    };
+    let target = env::var("TARGET").expect("TARGET was not set");
     let mut base_config = cc::Build::new();
     base_config
         .cpp(true)
-        .include("bitcoin/src")
-        .include("bitcoin/src/secp256k1/include")
-        .opt_level(2)
-        .debug(true)
```

Comment:
Removed these because the `cc` crate autodetects this and passes the same debug/release and opt level as rust got
https://github.com/rust-bitcoin/rust-secp256k1/pull/166